sklearn/cluster/_affinity_propagation.py

```diff
@@ -346,6 +362,13 @@ def __init__(self, *, damping=.5, max_iter=200, convergence_iter=15,
         self.verbose = verbose
         self.preference = preference
         self.affinity = affinity
+        if random_state == 'warning':
+            random_state = 0
+            warnings.warn(("'random_state' has been introduced in 0.23. "
+                           "It could be manually set. "
+                           "Its value is defaulted to 0, will be 'None' "
+                           "from 0.25."), FutureWarning)
+        self.random_state = random_state
```

Comment:
you need to do the validation in `fit`, and not in `__init__`. You can simply move this bit to `fit` and it should make the tests pass.

Reply:
Thanks @adrinjalali : I was convinced that the parameter should be set as 'warning' in the same function/class where the warning is... it shouldn't ...

Reply:
> you need to do the validation in `fit`, and not in `__init__`. You can simply move this bit to `fit` and it should make the tests pass.

Oh so sorry... this was written in the [doc](https://scikit-learn.org/stable/developers/contributing.html?highlight=docstring#change-the-default-value-of-a-parameter)! I've earned an RTFM today... this documentation is really good! :heart:

Reply:
it's a sad truth that we all realize it's in the doc once we need it and do it wrong the first time lol, and we have no idea how to fix it :D